File: leet_BS_153_min_rotated_sorted_arr___leftpointer.cpp
class Solution {
public:
    int findMin(vector<int>& n)
    {
        int len, R, L, mid ;

        len = n.size();
        R = len - 1;
        L = 0;
        if (len == 1 || n[0] < n[R])
            return n[0] ;
        while (L <= R)
        {
            mid = (R - L) / 2 + L;
            if (n[mid] >= n[len - 1] && n[mid] >= n[0])
                L = mid + 1;
            else
                R = mid - 1;
        }
        return (n[L]) ;
    }
};

File: leet_BS_153_min_rotated_sorted_arr___leftpointer.js
/**
 * @param {number[]} nums
 * @return {number}
 */

var findMin = function(arr) {
    len = arr.length
    if (len == 1)
        return arr[0]
    if (arr[0] < arr[len - 1])
        return arr[0]
    R = len - 1
    L = 0
    while (L < R) {
        mid = Math.floor((R - L) / 2) + L
        if (arr[mid] >= arr[len - 1] && arr[mid] >= arr[0])
            L = mid + 1
        else
            R = mid
    }
    return arr[L]
};
